Destiny.ScrimTracker.Api/Controllers/MatchController.cs

```diff
@@ -0,0 +1,131 @@
+        public IActionResult Index()
+        {
+            var matchResults = _matchService.GetMatchResults();
+            return View(matchResults);
+        }
+
+        [Authorize]
+        [Route("match")]
+        public IActionResult AddMatchView()
+        {
+            return View();
+        }
+        
+        [Authorize]
+        [Route("add_match")]
+        public IActionResult CreateMatchForm([FromQuery] int numOfTeams, [FromQuery] int playersPerTeam)
+        {
+            var guardians = _guardianService.GetGuardians();
+            ViewData.Add("Guardians", guardians.OrderBy(g => g.Guardian.GamerTag).Select(g => g.Guardian.GamerTag));
+            ViewData.Add("NumberOfTeams", numOfTeams);
+            ViewData.Add("PlayersPerTeam", playersPerTeam);
+
+            var match = new CreateMatchFormModel();
+
+            var teams = new List<Team>();
+            for (var i = 0; i < numOfTeams; i++)
+            {
+                teams.Add(new Team());
+                teams[i].MatchResults = new List<MatchResult>();
+                for (var j = 0; j < playersPerTeam; j++)
+                {
+                    teams[i].MatchResults.ToList().Add(new MatchResult());
+                }
+            }
+
+            match.Teams = teams;
+            return View(match);
+        }
+
+        [HttpPost]
+        [Authorize]
+        [Route("add_match")]
+        public IActionResult FormatRequestAndAddMatch([FromQuery] int numOfTeams, [FromQuery] int playersPerTeam)
+        {
+            var createMatchRequest = new CreateMatchRequest();
+
+            createMatchRequest.MatchType = Enum.Parse<MatchType>(Request.Form["MatchType"]);
+            var teams = new List<MatchTeam>();
+            for (var i = 0; i < numOfTeams; i++)
+            {
+                var team = new MatchTeam();
+                team.Name = Request.Form[$"Team[{i}].Name"];
+                team.TeamScore = int.Parse(Request.Form[$"Team[{i}].TeamScore"]);
+                
+                var guardianResults = new List<GuardianMatchResult>();
+                for (var j = 0; j < playersPerTeam; j++)
+                {
+                    var results = new GuardianMatchResult
+                    {
+                        Kills = int.Parse(Request.Form[$"Team[{i}].MatchResults[{j}].Kills"]),
+                        Assists = int.Parse(Request.Form[$"Team[{i}].MatchResults[{j}].Assists"]),
+                        Deaths = int.Parse(Request.Form[$"Team[{i}].MatchResults[{j}].Deaths"]),
+                        GuardianName = Request.Form[$"Team[{i}].MatchResults[{j}].GuardianName"]
+                    };
+
+                    if (results.Deaths == 0)
+                    {
+                        results.Efficiency = results.Kills + results.Assists;
+                    }
+                    else
+                    {
+                        results.Efficiency = (double) (results.Kills + results.Assists) / results.Deaths;
+                    }
```

Comment:
Does the ORM we're using have on save handlers? If yes, I'd prefer this logic lived there.

Reply:
Ah yea. Good call. I can push this logic a bit down the pipe

Reply:
Since I did a pretty large rework of this controller, I will also address this in the other PR. Will copy comment over to that one if its okay with ya?